apisix/cli/ngx_tpl.lua

```diff
@@ -719,7 +722,7 @@ http {
             set $upstream_connection         '';
 
             set $upstream_scheme             'http';
-            set $upstream_host               $http_host;
+            set $upstream_host               $host;
```

Comment:
The value of variable $http_host header will be empty string when using http/3
ref: zulip/zulip#26348